worlds/saving_princess/Options.py

```diff
@@ -0,0 +1,119 @@
+    With Jacket: you will not be able to sprint until after the Jacket item has been found.
+    """
+    display_name = "Sprint Availability"
+    option_never_available = 0
+    option_always_available = 1
+    option_available_with_jacket = 2
+    default = option_available_with_jacket
+
+
+class CliffWeaponUpgrade(Choice):
+    """
+    Determines which weapon Cliff uses against you, base or upgraded.
+    This does not change the available strategies all that much.
+    Vanilla: Cliff adds fire to his grenades if Ace has been defeated.
+    If playing with the expanded pool, the Arctic Key will trigger the change instead.
+    """
+    display_name = "Cliff Weapon Upgrade"
+    option_never_upgraded = 0
+    option_always_upgraded = 1
+    option_vanilla = 2
+    default = option_always_upgraded
+
+
+class AceWeaponUpgrade(Choice):
+    """
+    Determines which weapon Ace uses against you, base or upgraded.
+    Ace with his base weapon is very hard to dodge, the upgraded weapon offers a more balanced experience.
+    Vanilla: Ace uses ice attacks if Cliff has been defeated.
+    If playing with the expanded pool, the Volcanic Key will trigger the change instead.
+    """
+    display_name = "Ace Weapon Upgrade"
+    option_never_upgraded = 0
+    option_always_upgraded = 1
+    option_vanilla = 2
+    default = option_always_upgraded
+
+
+class ScreenShakeIntensity(Range):
+    """
+    Percentage multiplier for screen shake effects.
+    0% means the screen will not shake at all.
+    100% means the screen shake will be the same as in vanilla.
+    """
+    display_name = "Screen Shake Intensity %"
+    range_start = 0
+    range_end = 100
+    default = 50
+
+
+class IFramesDuration(Range):
+    """
+    Percentage multiplier for Portia's invincibility frames.
+    0% means you will have no invincibility frames.
+    100% means invincibility frames will be the same as vanilla.
+    """
+    display_name = "IFrame Duration %"
+    range_start = 0
+    range_end = 400
+    default = 100
+
+
+class TrapChance(Range):
+    """
+    Likelihood of a filler item becoming a trap.
+    """
+    display_name = "Trap Chance"
+    range_start = 0
+    range_end = 100
+    default = 50
+
+
+class MusicShuffle(Toggle):
+    """
+    Enables music shuffling.
+    The title screen song is not shuffled, as it plays before the client connects.
+    """
+    display_name = "Music Shuffle"
+
+
+@dataclass
+class SavingPrincessOptions(PerGameCommonOptions):
+    death_link: DeathLink
+    start_inventory_from_pool: StartInventoryPool
+    expanded_pool: ExpandedPool
+    instant_saving: InstantSaving
+    sprint_availability: SprintAvailability
+    cliff_weapon_upgrade: CliffWeaponUpgrade
+    ace_weapon_upgrade: AceWeaponUpgrade
+    shake_intensity: ScreenShakeIntensity
+    iframes_duration: IFramesDuration
+    trap_chance: TrapChance
+    music_shuffle: MusicShuffle
```

Comment:
```suggestion
@dataclass
class SavingPrincessOptions(PerGameCommonOptions):
    start_inventory_from_pool: StartInventoryPool
    expanded_pool: ExpandedPool
    trap_chance: TrapChance
    death_link: DeathLink
    instant_saving: InstantSaving
    sprint_availability: SprintAvailability
    cliff_weapon_upgrade: CliffWeaponUpgrade
    ace_weapon_upgrade: AceWeaponUpgrade
    shake_intensity: ScreenShakeIntensity
    iframes_duration: IFramesDuration
    music_shuffle: MusicShuffle
```

The order of the options in your dataclass determines what order the options show up in the player yamls and what order they show up in on the website. I recommend grouping the "generation" options together towards the top and the "client" options towards the bottom. The suggestion is my recommendation here.